collect_app/src/main/java/org/odk/collect/android/logic/FormController.java

```diff
@@ -503,18 +510,14 @@ public int stepToNextEvent(boolean stepIntoGroup) {
      * the group represented by the FormIndex.
      */
     private int stepOverGroup() {
-        ArrayList<FormIndex> indicies = new ArrayList<FormIndex>();
+        ArrayList<FormIndex> indicies = new ArrayList<>();
         GroupDef gd =
                 (GroupDef) formEntryController.getModel().getForm()
                         .getChild(getFormIndex());
         FormIndex idxChild =
                 formEntryController.getModel().incrementIndex(
                         getFormIndex(), true); // descend into group
-        for (int i = 0; i < gd.getChildren().size(); i++) {
-            indicies.add(idxChild);
-            // don't descend
-            idxChild = formEntryController.getModel().incrementIndex(idxChild, false);
-        }
+        indicies.addAll(getIndices(gd, idxChild));
 
         // jump to the end of the group
         formEntryController.jumpToIndex(indicies.get(indicies.size() - 1));
```

Comment:
Why add the result of getIndices to a second List? I suggest a simplification. Note the correct spelling of *indices*, and use of the more general `List` instead of `ArrayList`.

![screen shot 2018-02-07 at 3 19 08 pm](https://user-images.githubusercontent.com/55099/35946921-7d70343c-0c1a-11e8-9711-a11f2a1435ef.png)